### ISSUE TITLE
Python wrapper for the sparse case

### ISSUE THREAD

**Issue by ilyaraz** [FALCONN-LIB/FALCONN#24]
Currently, the Python wrapper supports only dense data (via Numpy arrays), would be nice to generalize it to handle sparse data as well.

**Comment by ludwigschmidt** [comment]
Yes, eventually we should support sparse data from python. There are three points we should figure out first:
- What is the sparse vector format we want to use on the C++? See issue #5 .
- What is the sparse vector format we want to use on the python side? A natural choice might be http://docs.scipy.org/doc/scipy/reference/sparse.html .
- How to connect the two sparse vector formats, ideally without copying data (similar to the dense vector case). The official swig numpy wrapper doesn't seem to support scipy's sparse vectors (at least it is not documented here: http://docs.scipy.org/doc/numpy-1.10.0/reference/swig.interface-file.html ).
